### ISSUE TITLE
Failed to resolve entry for package when use react + vite

### ISSUE THREAD

**Issue by dfiai** [nrwl/nx#18931]
### Current Behavior

I create 2 packages app and lib using react/vite, I got build app error when import lib in to the app.

### Expected Behavior

Lib should be referenced in the app without errors.

### Steps to Reproduce

1. create 2 packages using nx/react/vite app and lib
![image](https://github.com/nrwl/nx/assets/139440179/ecb4d6da-85b9-4fec-9816-f2e07179d567)

2. add import lib in the app
![image](https://github.com/nrwl/nx/assets/139440179/39e959ad-0375-42f8-8fc1-f580140204fb)

3. `npx nx build my-app`: [commonjs--resolver] Failed to resolve entry for package "my-lib". The package may have incorrect main/module/exports specified in its package.json.

It was reproducible in ubuntu and windows , magically after renaming one of the package it started to work but after `npm i`, it stopped again

### Nx Report

```shell
Node   : 16.16.0
   OS     : win32-x64
   npm    : 8.11.0

   nx                 : 16.7.4
   @nx/js             : 16.7.4
   @nx/jest           : 16.7.4
   @nx/linter         : 16.7.4
   @nx/workspace      : 16.7.4
   @nx/devkit         : 16.7.4
   @nx/eslint-plugin  : 16.7.4
   @nx/react          : 16.7.4
   @nrwl/tao          : 16.7.4
   @nx/vite           : 16.7.4
   @nx/web            : 16.7.4
   typescript         : 5.1.6
```

### Package Manager Version

8.11.0

### Operating System

- [ ] macOS
- [X] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by steven10172** [comment]
Related to https://github.com/nrwl/nx/issues/18492#issuecomment-1691942830

**Comment by dbehmoaras** [comment]
I am running into the exact same issue

**Comment by MBelniak** [comment]
Have anyone tried enabling this option? https://vitejs.dev/config/shared-options#resolve-preservesymlinks
It helped in my case (the path to main was being referenced from resolved location instead of path in node_modules)

**Comment by Coly010** [comment]
Hi! Could you please provide a repo with a minimal reproduction so that I can investigate this further?

**Comment by Wurielle** [comment]
> Has anyone tried enabling this option? https://vitejs.dev/config/shared-options#resolve-preservesymlinks It helped in my case (the path to main was being referenced from resolved location instead of path in node_modules)

I just tried this but it didn't work for me although I found what's causing the issue in the process.

The issue only appeared when I added `workspaces` in my root `package.json` file.
Something about Vite aliases generated by `nxViteTsPaths` + the fact that the packages are already present within `node_modules` (due to workspaces) breaks the module resolution for Vite.

I haven't found a solution for my problem except by manually setting `resolve.alias` in my `vite.config.ts` for each package.
Setting `resolve.preserveSymlinks = true` doesn't change anything for me.
